### PR TITLE
[move-lang][error migration] Migrate expansion/ errors

### PR DESCRIPTION
**Pull request by tnowacki**
- Migrate expansion errors
- Update cycle errors to point to all locations (hopefully that is better?)
- Removed a few useless tests

## Motivation

- More progress on migrating errors

## Test Plan

- Updated tests

### CODE REVIEW

File: language/move-lang/tests/move_check/dependencies/friend_cycle_2.exp [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+  ┌─ tests/move_check/dependencies/friend_cycle_2.move:8:5
+  │
+4 │     friend 0x2::B;
+  │     -------------- '0x2::B' is a friend of '0x2::A'
+  ·
+8 │     friend 0x2::A;
+  │     ^^^^^^^^^^^^^^
+  │     │
+  │     Invalid friend '0x2::A' in module '0x2::B'. This 'friend' relationship creates a dependency cycle.
+  │     '0x2::A' is a friend of '0x2::B'
```
Comment (mengxu-fb) [inline language/move-lang/tests/move_check/dependencies/friend_cycle_2.exp:11]: I like this new style of messages regarding dependency cycles!

Just one minor suggestion: would it be better to swap line 10 and 11? Say,
```
error[E02004]: invalid 'module' declaration
  ┌─ tests/move_check/dependencies/friend_cycle_2.move:8:5
  │
4 │     friend 0x2::B;
  │     -------------- '0x2::B' is a friend of '0x2::A'
  ·
8 │     friend 0x2::A;
  │     ^^^^^^^^^^^^^^
  │     │
  |     '0x2::A' is a friend of '0x2::B'. This 'friend' relationship creates a dependency cycle.
```

Comment (tnowacki) [reply language/move-lang/tests/move_check/dependencies/friend_cycle_2.exp]: I changed it a bit, let me know what you think

Comment (mengxu-fb) [reply language/move-lang/tests/move_check/dependencies/friend_cycle_2.exp]: LGTM!

Comment (tnowacki) [reply language/move-lang/tests/move_check/dependencies/friend_cycle_2.exp]: Could you approve it if everything looks good? :)

**Review by mengxu-fb — DISMISSED** [review]
Sorry for the late approval

**Comment by tnowacki** [comment]
Manually squashed hoping it would fix the erroneous merge conflict (since I got nothing when rebasing)

**Review by mengxu-fb — DISMISSED** [review]

**Pull request closed without merging**